docs/Custom-brimcap-load-Config.md

```diff
@@ -0,0 +1,433 @@
+to match the `ts` one used by Zeek, which allows the data from both sources to
+be more seamlessly presented together in Brim.
+
+# Custom NetFlow Analyzer
+
+## Background
+
+Building on what we just learned about Brimcap analyzers, we can extend the
+concepts to pcap processing tools other than Zeek or Suricata. In this case
+we'll use NetFlow records generated using the open source
+[nfdump](https://github.com/phaag/nfdump) toolset.
+
+* **Note**: While the example shows how shaped NetFlow records can be loaded
+via Brimcap, the **Packets** button in Brim is not currently wired to extract
+flows from a pcap via the 5-tuple and time details in NetFlow records. If this
+is functionality you're interested in pursuing, please
+[open an issue](https://github.com/brimdata/brimcap/wiki/Troubleshooting#opening-an-issue)
+or come talk to us on [public Slack](https://www.brimsecurity.com/join-slack/).
+
+## Base nfdump Installation
+
+To build and install the nfdump toolset with the commands we'll need to
+generate NetFlow, we execute the following on our Linux host:
+
+```
+sudo apt install automake libtool pkg-config flex bison libbz2-dev
+git clone https://github.com/phaag/nfdump.git && cd nfdump
+./autogen.sh
+./configure --enable-nfpcapd
+sudo make install
+export LD_LIBRARY_PATH="/usr/local/lib"
+```
+
+## Example Configuration
+
+As we did with Zeek and Suricata, we create a wrapper script to act as our
+Brimcap analyzer. It works in two phases, first creating binary NetFlow records
+and then converting them to CSV. `nfpcapd` only accepts a true pcap file input
+(not a device like `/dev/stdin`), so we first store the incoming pcap in a
+temporary file. We also use GNU `head` to trim out some trailing summary lines
+that `nfdump` appends by default which would trip up the Zed CSV reader.
+
+```
+$ cat nfdump-wrapper.sh 
+TMPFILE=$(mktemp)
+cat - > "$TMPFILE"
+nfpcapd -r "$TMPFILE" -l .
+rm "$TMPFILE"
+for file in nfcapd.*
+do
+  nfdump -r $file -o csv | head -n -3 | zq -i csv -f ndjson - > ${file}.ndjson
+done
+```
+
+This script is called from our Brimcap config YAML, which includes a
+`globs:` setting to target only the CSV files and also a Zed shaper to apply
+rich data types.
+
+```
+$ cat nfdump.yml 
+analyzers:
+  - cmd: nfdump-wrapper.sh
+    # The globs being set to "*.ndjson" is a workaround for the fact that our
+    # current JSON reader won't accept much data and we can't tell
+    # brimcap load to expect CSV input, so we're postprocessing in the
+    # nfdump-wrapper.sh script with zq to turn the CSV back into NDJSON.
+    globs: ["*.ndjson"]
+    shaper: |
+      type netflow = {
+        ts: time,
+        te: time,
+        td: duration,
+        sa: ip,
+        da: ip,
+        sp: uint16,
+        dp: uint16,
+        pr: string,
+        flg: string,
+        fwd: bytes,
+        stos: bytes,
+        ipkt: uint64,
+        ibyt: uint64,
+        opkt: uint64,
+        obyt: uint64,
+        \in: uint64,
```

Comment:
Note the escape because `in` is reserved. I'm not sure if brimdata/zed#2398 will address this, but I hope so because stumbling onto these clashes with reserved words while working with new data sources is not much fun.

Reply:
As I'm not sure if brimdata/zed#2398 will make the cut for MVP0, for now I've added a "Note:" block linking to the open issue.